File: test_glm.js
#!/usr/bin/env node
/**
 * 测试智谱AI API连接 (Node.js)
 */

// 检查环境变量
const apiKey = process.env.ANTHROPIC_API_KEY;
const apiBase = process.env.ANTHROPIC_API_BASE;

if (!apiKey) {
    console.log('❌ 错误：未找到 ANTHROPIC_API_KEY 环境变量');
    console.log('请先运行: source .env 或使用 glm 命令');
    process.exit(1);
}

console.log(`🔑 API Key: ${apiKey.substring(0, 20)}...`);
console.log(`🌐 API Base: ${apiBase}`);
console.log('\n✅ 环境变量加载成功！');
console.log('\n📝 注意：实际 API 调用需要安装 @anthropic-ai/sdk：');
console.log('   npm install @anthropic-ai/sdk');
console.log('\n示例代码：');
console.log(`
const Anthropic = require('@anthropic-ai/sdk');

const client = new Anthropic({
    apiKey: process.env.ANTHROPIC_API_KEY,
    baseURL: process.env.ANTHROPIC_API_BASE
});

// 使用 client 进行 API 调用
`);

File: test_glm.py
#!/usr/bin/env python3
"""
测试智谱AI API连接
"""
import os
from zhipuai import ZhipuAI

# 从环境变量读取API密钥
api_key = os.environ.get("ANTHROPIC_API_KEY")
api_base = os.environ.get("ANTHROPIC_API_BASE")

if not api_key:
    print("❌ 错误：未找到 ANTHROPIC_API_KEY 环境变量")
    print("请先运行: source .env")
    exit(1)

print(f"🔑 API Key: {api_key[:20]}...")
print(f"🌐 API Base: {api_base}")
print("\n正在测试连接...")

try:
    client = ZhipuAI(api_key=api_key)

    # 测试一个简单的调用
    response = client.chat.completions.create(
        model="glm-4-flash",  # 使用快速模型进行测试
        messages=[
            {"role": "user", "content": "你好，请用一句话介绍你自己"}
        ],
        max_tokens=100
    )

    print("✅ 连接成功！")
    print(f"\n回复: {response.choices[0].message.content}")

except Exception as e:
    print(f"❌ 连接失败: {str(e)}")
    exit(1)
